lib/rspec/rails/matchers/active_job.rb

```diff
@@ -230,11 +230,26 @@ def initialize(job)
           def matches?(proc)
             raise ArgumentError, "have_enqueued_job and enqueue_job only support block expectations" unless Proc === proc
 
-            original_enqueued_jobs_count = queue_adapter.enqueued_jobs.count
+            original_enqueued_jobs_hashes = queue_adapter.enqueued_jobs.map(&:hash)
+
             proc.call
-            in_block_jobs = queue_adapter.enqueued_jobs.drop(original_enqueued_jobs_count)
 
-            check(in_block_jobs)
+            in_block_jobs = queue_adapter.enqueued_jobs.each_with_object({}) do |job, memo|
```

Comment:
Done.
I tried to name it more clearly and failed. IMHO `jobs` is not ideal because everything is about jobs in this method and it's not clear what `jobs` we're talking about, but I guess it's better than `memo`.